ts/data.ts

```diff
@@ -1,8 +1,29 @@
 /* exported data */
 
-let data = {
-  view: 'entry-form',
+interface JournalEntry {
+  entryId: number;
+  title: string;
+  photoUrl: string;
+  notes: string;
+}
+
+interface DataModel {
+  entries: JournalEntry[];
+  nextEntryId: number;
+}
+
+const data: DataModel = {
   entries: [],
-  editing: null,
   nextEntryId: 1,
 };
+
+const savedData = localStorage.getItem('code-journal-data');
+if (savedData) {
+  Object.assign(data, JSON.parse(savedData));
+}
+
```

Comment:
Where did you get this `Object.assign()` from? We can just use the assignment operator like we did in the `typescript-local-storage` exercise `data = JSON.parse(savedData)`